Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -3064,27 +3088,110 @@ namespace OpenInfraPlatform
 					else if (polynomialConstantCntX>0 && polynomialConstantCntY>0 && polynomialConstantCntZ>0) return carve::geom::VECTOR(x, y, z);
 				}
 
-				/*! \brief Calculates a polinomial curve.
+				/*! \brief Calculates a polinomial curve
 				* \param[in] coefficient			A vector with coeffitients corresponding ordinate from \c IfcPolynomialCurve.
 				* \param[in] parameter				A length during the curve from \c IfcCurveSegment.
 				* \return							The location of the ordinate point.
 				* \note
 				*/
-				typename IfcEntityTypesT::IfcReal calculatePolynomialCurve(const std::vector<typename IfcEntityTypesT::IfcReal> coefficient, double length) const noexcept(false)
+				typename IfcEntityTypesT::IfcReal calculatePolynomialCurve(std::vector<double> coefficient, const double parameter) const noexcept(false)

```

Comment:
```suggestion
				typename IfcEntityTypesT::IfcReal calculatePolynomialCurve(std::vector<double>& coefficient, const double parameter) const noexcept(false)
```

If you're not changing the values of the array, always pass it by reference, not by value (saves time and RAM).

Reply:
Thank you for your advice! It is really faster than usual.
Resolved: 98adeb1